extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -457,6 +456,31 @@ export class SubscriptionService {
       platform.decorateUI(element, action, options);
     });
   }
+
+  /**
+   * Evaluates platforms and select the one to be selected for login.
+   * @return {!Promise<!./subscription-platform.SubscriptionPlatform>}
+   */
+  scoreBasedLogin() {
+    return this.platformStore_.getAllPlatforms().then(platforms => {
+      const platformScores = [];
+      platforms.forEach(platform => {
+        let score = 0;
+        if (platform.supportsCurrentViewer()) {
+          score += 1000;
+        }
+        platformScores.push({
+          platform,
+          score,
+        });
+      });
+
+      platformScores.sort(function(platform1, platform2) {
+        return platform2.weight - platform1.weight;
+      });
+      return platformScores[0].platform;
```

Comment:
If none of the platforms matches viewer - would this guarantee that local wins then?

Reply:
Do we have a reason to do this?

Reply:
Yes. With all equal "local" should win.